File: Samples/azure-storage/Azure.Python/storagemanagementclient/models/endpoints.py
# coding=utf-8
# --------------------------------------------------------------------------
# --------------------------------------------------------------------------

from msrest.serialization import Model


class Endpoints(Model):
    """The URIs that are used to perform a retrieval of a public blob, queue or
    table object.

    :param blob: Gets the blob endpoint.
    :type blob: str
    :param queue: Gets the queue endpoint.
    :type queue: str
    :param table: Gets the table endpoint.
    :type table: str
    :param file: Gets the file endpoint.
    :type file: str
    """ 

    _attribute_map = {
        'blob': {'key': 'blob', 'type': 'str'},
        'queue': {'key': 'queue', 'type': 'str'},
        'table': {'key': 'table', 'type': 'str'},
        'file': {'key': 'file', 'type': 'str'},
    }

    def __init__(self, blob=None, queue=None, table=None, file=None):
        self.blob = blob
        self.queue = queue
        self.table = table
        self.file = file

File: Samples/azure-storage/Azure.CSharp/Models/Endpoints.cs

namespace Petstore.Models
{
    using Newtonsoft.Json;
    using System.Linq;

    /// <summary>
    /// The URIs that are used to perform a retrieval of a public blob, queue
    /// or table object.
    /// </summary>
    public partial class Endpoints
    {
        /// <summary>
        /// Initializes a new instance of the Endpoints class.
        /// </summary>
        public Endpoints() { }

        /// <summary>
        /// Initializes a new instance of the Endpoints class.
        /// </summary>
        /// <param name="blob">Gets the blob endpoint.</param>
        /// <param name="queue">Gets the queue endpoint.</param>
        /// <param name="table">Gets the table endpoint.</param>
        /// <param name="file">Gets the file endpoint.</param>
        public Endpoints(string blob = default(string), string queue = default(string), string table = default(string), string file = default(string))
        {
            Blob = blob;
            Queue = queue;
            Table = table;
            File = file;
        }

        /// <summary>
        /// Gets the blob endpoint.
        /// </summary>
        [JsonProperty(PropertyName = "blob")]
        public string Blob { get; set; }

        /// <summary>
        /// Gets the queue endpoint.
        /// </summary>
        [JsonProperty(PropertyName = "queue")]
        public string Queue { get; set; }

        /// <summary>
        /// Gets the table endpoint.
        /// </summary>
        [JsonProperty(PropertyName = "table")]
        public string Table { get; set; }

        /// <summary>
        /// Gets the file endpoint.
        /// </summary>
        [JsonProperty(PropertyName = "file")]
        public string File { get; set; }

    }
}

